var/spack/repos/builtin/packages/cdo/package.py

```diff
@@ -155,6 +155,10 @@ class Cdo(AutotoolsPackage):
     depends_on("cxx", type="build")  # generated
     depends_on("fortran", type="build")  # generated
 
+    # patches
+    # see https://code.mpimet.mpg.de/boards/1/topics/15594
+    patch("add_algorithm_header.patch", when="@2.4.0 %gcc@14")
```

Comment:
Is this patch not needed for 2.3.0 or other older versions? Do older versions compile with gcc@14?

I'd imagine this could look like
```suggestion
    patch("add_algorithm_header.patch", when="@2.2:2.4.0 %gcc@14:")
    conflicts("%gcc@14:", when="@:2.1", msg="Compilation with gcc@14: requires cdo@2.2:")
```
Note: entirely made up versions as example only.

Reply:
not 100% sure atm - will run a couple of test on that

Reply:
send u patch via pm. works on my system with gcc14 from 2.1.0 till 2.4.0

Reply:
Thank you, I applied your patch and added the `conflicts` as well, since old versions still have troubles. As a drawback of `conflicts` on my system spack then reuses dependencies built with gcc and tries to build cdo with my clang, which again fails. But since #46033 added new versions of cdo without any problem, I would disregard this drawback